src/RecordComponent.cpp

```diff
@@ -176,6 +178,39 @@ RecordComponent::empty() const
     return *m_isEmpty;
 }
 
+void
+RecordComponent::serialize()
+{
+    // verify: read-only
+    if( IOHandler()->m_frontendAccess != Access::READ_ONLY )
+        throw std::runtime_error("serialize only works on read-only access");
+    //if( IOHandler()->m_backendAccess != Access::READ_ONLY )
+    // as well??
```

Comment:
Ref.: https://github.com/openPMD/openPMD-api/pull/460
Tldr: It should not matter at this point